src/osvr/Client/RenderManagerConfig.cpp

```diff
@@ -0,0 +1,190 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/Client/RenderManagerConfig.h>
+#include <osvr/Common/ClientContext.h>
+#include <osvr/Util/Verbosity.h>
+
+// Library/third-party includes
+#include <json/value.h>
+#include <json/reader.h>
+
+namespace osvr {
+namespace client {
+
+    RenderManagerConfigPtr RenderManagerConfigFactory::create(OSVR_ClientContext ctx) {
+        try {
+            auto const configString = ctx->getStringParameter("/display");
+            RenderManagerConfigPtr cfg(new RenderManagerConfig(configString));
+            return cfg;
+        }
+        catch (std::exception const &e) {
+            OSVR_DEV_VERBOSE(
+                "Couldn't create a render manager config internally! Exception: "
+                << e.what());
+            return RenderManagerConfigPtr{};
+        }
+        catch (...) {
+            OSVR_DEV_VERBOSE("Couldn't create a display config internally! "
+                "Unknown exception!");
+            return RenderManagerConfigPtr{};
+        }
+    }
+
+    RenderManagerConfig::RenderManagerConfig(const std::string& render_manager_config)
+    {
+        parse(render_manager_config);
+    }
+
+    void RenderManagerConfig::parse(const std::string& render_manager_config)
+    {
+      Json::Reader reader;
+      Json::Value root;
+      reader.parse(render_manager_config, root, false);
+
+      m_directMode = root["render_manager_parameters"]["direct_mode"].asBool();
+      m_displayIndex = root["render_manager_parameters"]["direct_display_index"].asUInt();
+      m_directHighPriority = root["render_manager_parameters"]["direct_high_priority"].asBool();
+
+      m_numBuffers = root["render_manager_parameters"]["num_buffers"].asUInt();
+      m_verticalSync = root["render_manager_parameters"]["vertical_sync"].asBool();
+      m_verticalSyncBlockRendering = root["render_manager_parameters"]
+        ["vertical_sync_block_rendering"].asBool();
+      m_windowTitle = root["render_manager_parameters"]["window_title"].asString();
+      m_windowFullScreen = root["render_manager_parameters"]["window_full_screen"].asBool();
+      m_windowXPosition = root["render_manager_parameters"]["window_x_position"].asInt();
+      m_windowYPosition = root["render_manager_parameters"]["window_y_position"].asInt();
+      m_displayRotation = root["render_manager_parameters"]["display_rotation"].asUInt();
+      m_bitsPerColor = root["render_manager_parameters"]["bits_per_color"].asBool();
+
+      m_enableTimeWarp = root["render_manager_parameters"]["time_warp"]["enable"].asBool();
+      m_asynchronousTimeWarp = root["render_manager_parameters"]["time_warp"]["asynchronous_time_warp"].asBool();
+      m_maxMSBeforeVsyncTimeWarp = root["render_manager_parameters"]["time_warp"]["max_ms_before_vsync"].asFloat();
```

Comment:
Here I can actually show a bit better how my don't-repeat-yourself-accessor-style works, since there's actually something nested another level. The new scope is to limit the lifetime of the reference to just the code that needs it, keeps it clean and keeps the indentation matching the nesting level of the schema. (You can see a little of the cleanup I did on code that started like this in the display config stuff)

```c++
{
    auto & timeWarp = params["timeWarp"];
    m_enableTimeWarp = timeWarp["enable"].asBool();
    m_asynchronousTimeWarp = timeWarp["asynchronous"].asBool();
    m_maxMSBeforeVsyncTimeWarp = timeWarp["maxMillisecondsBeforeVsync"].asFloat();
}
```